mitodl/open-discussions
# Cover image for articles - backend

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1593 

#### What's this PR do?
- Add `cover_image` and `cover_image_small` fields to the `Article` model
- Updates the `PostSerializer` to handle the new fields (get/post/patch)
- Customizes `Article.save` method to autogenerate the thumbnail (`cover_image_small`) if the `cover_image` field is being updated.

#### How should this be manually tested?
There's no front-end yet but you can simulate creating an article post with cover image through the shell:

```python
import os
import json
from django.contrib.auth.models import User
from rest_framework.test import APIClient
from channels.models import Channel, Post, Article

client = APIClient()
client.force_login(User.objects.get(username='<username>'))
channel = Channel.objects.get(name="<channel_name>")
article_content = [{"key": "value", "nested": {"number": 4}}]
img_file = "static/images/mit-logo-micromasters.jpg"
url = reverse("post-list", kwargs={"channel_name": channel.name})
with open(img_file, "rb") as f:
    resp = client.post(
        url, 
        {"title": "foo", "article_content": json.dumps(article_content), "cover_image": f}, 
        format="multipart"
    )

post = Post.objects.get(post_id=resp.json()["id"])
print(post.article.cover_image.url, post.article.cover_image_small.url)
```

Then go to `http://od.odl.local:8063/api/v0/posts/<post_id>/` and verify that there are working URL's for `cover_image` and `thumbnail` that display the image file you used above.
